### PR TITLE
Security_Engine: Camera Field of View bug fix

### LINKED ISSUE #3265: Security_Engine: Camera Field Of View Bug Fix
#### Description:

- Method returns null when camera device is inside obstacles that could be just skipped.
- Arc creation crashes in some more complex cases

### PR DESCRIPTION
**Pull request by michal-pekacki**
### Issues addressed by this PR

Closes #3265

- arc creation bug fixed
- skip obstacles if camera device is inside (don't return null as before)

### CODE REVIEW

**Comment by michal-pekacki** [comment]
@BHoMBot check compliance

File: Security_Engine/Compute/CameraFieldOfView.cs [hunk 1]
```diff
@@ -129,7 +129,10 @@ public static PolyCurve CameraFieldOfView(this CameraDevice cameraDevice, List<P
             //create points chain
             List<Point> pointsChain = linesDict.PointsChain(cameraLocation, radius, distanceTolerance);
 
-            //create cone
+            //reverse points chain to have correct start and end arc angles
+            pointsChain.Reverse();
```
Comment (pawelbaran) [inline Security_Engine/Compute/CameraFieldOfView.cs:133]: Shouldn't that be a part of `Query.PointsChain` method?

**Review by pawelbaran — COMMENTED** [review]
Changes make sense in general, only one small question asked

File: Security_Engine/Compute/CameraFieldOfView.cs (hunk 1 above)
Comment (michal-pekacki) [reply Security_Engine/Compute/CameraFieldOfView.cs:133]: Yes, I will move that 👍

**Review by pawelbaran — APPROVED** [review]
No concerns code-wise, happy to see this merged if the workflows benefit 👍

**Comment by michal-pekacki** [comment]
@BHoMBot check compliance

**Comment by michal-pekacki** [comment]
@BHoMBot check required

**Comment by michal-pekacki** [comment]
@BHoMBot check unit-tests

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am authorising dispensation to be granted on check ref. 21146284981

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**